### PR TITLE
Add option to upload Jenkins test result to BQ

### PR DESCRIPTION
**Pull request by matt-kwong**
After this is merged, flags just need to be added to Jenkins master jobs. 

I tested this on a GCE Linux VM.

### CODE REVIEW

File: tools/run_tests/python_utils/upload_test_results.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      test_results['compiler'] = args.compiler
+      test_results['config'] = args.config
+      test_results['cpu_estimated'] = result.cpu_estimated
+      test_results['cpu_measured'] = result.cpu_measured
+      test_results['elapsed_time'] = '%.2f' % result.elapsed_time
+      test_results['iomgr_platform'] = args.iomgr_platform
+      # args.language is a list, but will always have one element in the contexts
+      # this function is used.
+      test_results['language'] = args.language[0]
+      test_results['passed'] = bool(result.state == 'PASSED')
```
Comment (ctiller) [inline tools/run_tests/python_utils/upload_test_results.py]: Could we get timeout, skipped rows also?

Comment (matt-kwong) [inline tools/run_tests/python_utils/upload_test_results.py]: AFAIK, if we're only uploading results from master jobs, there shouldn't be any skipped tests. I'll add a field for timeouts.

**Comment by ctiller** [comment]
I'd like to get this on PRs also... I suspect we'll find some uses for the
data... And there we do skip some tests.

**Comment by matt-kwong** [comment]
Gotcha, in that case I think it's better to change the `passed` field to `result` and make it a string.

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

**Comment by matt-kwong** [comment]
Tests were failing because `big_query_utils.py` being imported was causing errors w/ `run_tests.py` because dependencies weren't installed (they were only installed on perf VMs). I installed them on Linux, Windows, and Macs workers.

Also, the `result` field should include timeout and skipped, in addition to passed and failed.

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

File: tools/run_tests/python_utils/upload_test_results.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    os.path.dirname(__file__), '../../gcp/utils'))
+sys.path.append(gcp_utils_dir)
+import big_query_utils
+
+_DATASET_ID = 'jenkins_test_results'
+_DESCRIPTION = 'Test results from master job run on Jenkins'
+_PROJECT_ID = 'grpc-testing'
+_RESULTS_SCHEMA = [
+  ('build_id', 'INTEGER', 'Build ID of Jenkins job'),
+  ('build_url', 'STRING', 'URL of Jenkins job'),
```
Comment (ctiller) [inline tools/run_tests/python_utils/upload_test_results.py]: As a convenience, could we capture the Jenkins job here also (will make queries easier in the future).

**Comment by matt-kwong** [comment]
Since I changed the Dockerfiles, they are no longer being built from cache and a few unrelated failures have come up. They seem like trivial fixes, so I'm fixing them before merging.

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

**Comment by matt-kwong** [comment]
jenkins, test this please

**Comment by matt-kwong** [comment]
All failures are known issues and they were passing in previous runs. The "unrelated" issues I referred to earlier were actually my fault and those tests are now passing.

**Pull request merged**